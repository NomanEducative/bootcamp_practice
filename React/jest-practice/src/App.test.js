import { render, screen } from '@testing-library/react';
import Login from './App';
import Enzyme,{ shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'

Enzyme.configure({ adapter: new Adapter()})

describe('Matching Snapshots',()=>{
  it('Matches Snapshot',()=>{
    const tree=renderer.create(<Login/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
})

describe('App',()=>{
  it('Renders Without Crashing and All Labels Check',()=>{
    const wrapper=shallow(<Login/>);
    const login=wrapper.find('h2.firstHeading').text();
    expect(login).toEqual("Login Form");
    const email=wrapper.find('input#email').text();
    expect(email).toEqual("");
    const password=wrapper.find('input#password').text();
    expect(password).toEqual("");
  })

  it('Check change in Input Field Value',()=>{
    const wrapper=shallow(<Login/>);
    const firstField=wrapper.find('input#email');
    firstField.simulate('change',{ target: { value: 'Hello' } })
    const emailFromField=firstField.text();
    const stateEmail=wrapper.state().email;
    expect(stateEmail).toEqual(emailFromField)
    const secondField=wrapper.find('input#password');
    secondField.simulate('change')
    const passwordFromField=firstField.text();
    const statePassword=wrapper.state().email;
    expect(statePassword).toEqual(passwordFromField)
  })

  it('Check For Buttons Click',()=>{
    const wrapper=shallow(<Login/>);
    const button=wrapper.find('button');
    button.simulate('click');
    expect(wrapper.state().loginMessage).toEqual("You're now logged into the account " + wrapper.state().email + "!")
    
  })
})


