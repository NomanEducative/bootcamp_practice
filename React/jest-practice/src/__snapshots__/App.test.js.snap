// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matching Snapshots Matches Snapshot 1`] = `
<div>
  <h2
    className="firstHeading"
  >
    Login Form
  </h2>
  <form
    className="login"
  >
    <label
      forHtml="email"
    >
      Email
    </label>
    <input
      id="email"
      name="email"
      onBlur={[Function]}
      type="text"
    />
    <label
      className="passwordLabel"
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      onBlur={[Function]}
      type="password"
    />
    <button
      className="outlined-default"
      onClick={[Function]}
    >
      Submit
    </button>
  </form>
  <div
    id="loginMessage"
  >
    <h2
      className="message"
    />
  </div>
</div>
`;
